blueman/gui/manager/ManagerDeviceList.py

```diff
@@ -468,40 +420,26 @@ def row_update_event(self, tree_iter: Gtk.TreeIter, key: str, value: Any) -> Non
         elif key == "Blocked":
             self.set(tree_iter, blocked=value)
 
-    def _update_power_levels(self, tree_iter: Gtk.TreeIter, device: Device, cinfo: conn_info) -> None:
-        row = self.get(tree_iter, "cell_fader", "battery", "rssi", "lq", "tpl")
-
-        bars = {}
+        elif key == "RSSI":
+            self._update_bar(tree_iter, "rssi", 50 if value is None else max(50 + float(value) / 127 * 50, 10))
 
-        obj_path = device.get_object_path()
-        if obj_path in self._batteries:
-            bars["battery"] = self._batteries[obj_path]["Percentage"]
+        elif key == "TxPower":
+            self._update_bar(tree_iter, "tpl", 0 if value is None else max(float(value) / 127 * 100, 10))
 
-        # cinfo init may fail for bluetooth devices version 4 and up
-        # FIXME Workaround is horrible and we should show something better
-        if cinfo.failed:
-            bars.update({"rssi": 100.0, "tpl": 100.0})
-        else:
-            try:
-                bars["rssi"] = max(50 + float(cinfo.get_rssi()) / 127 * 50, 10)
-            except ConnInfoReadError:
-                bars["rssi"] = 50
-            try:
-                bars["tpl"] = max(50 + float(cinfo.get_tpl()) / 127 * 50, 10)
-            except ConnInfoReadError:
-                bars["tpl"] = 50
+    def _update_bar(self, tree_iter: Gtk.TreeIter, name: str, perc: float) -> None:
+        row = self.get(tree_iter, "cell_fader", "battery", "rssi", "tpl")
+        row[name] = perc
 
         if row["battery"] == row["rssi"] == row["tpl"] == 0:
             self._prepare_fader(row["cell_fader"]).animate(start=0.0, end=1.0, duration=400)
 
         w = 14 * self.get_scale_factor()
         h = 48 * self.get_scale_factor()
 
-        for (name, perc) in bars.items():
-            if round(row[name], -1) != round(perc, -1):
-                icon_name = f"blueman-{name}-{int(round(perc, -1))}.png"
-                icon = GdkPixbuf.Pixbuf.new_from_file_at_scale(os.path.join(PIXMAP_PATH, icon_name), w, h, True)
-                self.set(tree_iter, **{name: perc, f"{name}_pb": icon})
+        if round(row[name], -1) != round(perc, -1):
+            icon_name = f"blueman-{name}-{int(round(perc, -1))}.png"
+            icon = GdkPixbuf.Pixbuf.new_from_file_at_scale(os.path.join(PIXMAP_PATH, icon_name), w, h, True)
+            self.set(tree_iter, **{name: perc, f"{name}_pb": icon})
```

Comment:
Somewhere you will need to `_disable_power_levels` otherwise the bar will remain forever.

Reply:
:thinking: Don't we rather need to handle when the properties get invalidated?